go-controller/pkg/clustermanager/zone_cluster_controller.go

```diff
@@ -131,6 +131,23 @@ func (zcc *zoneClusterController) Stop() {
 	}
 }
 
+func needsZoneAllocation(node *corev1.Node) bool {
+	if config.HybridOverlay.Enabled && util.NoHostSubnet(node) {
+		// skip hybrid overlay nodes
+		return false
+	}
+
+	if _, ok := node.Annotations[util.OvnNodeID]; !ok {
+		return true
+	}
+	if config.OVNKubernetesFeature.EnableInterconnect {
+		if _, ok := node.Annotations[util.OvnTransitSwitchPortAddr]; !ok {
+			return true
+		}
+	}
```

Comment:
doesn't this part also mean that the node will be in `nodeSyncFailed` map?

Reply:
that's checked where this function is called:
```
		if !nodeFailed && !needsZoneAllocation(node) {
			// node ID and transit switch IP are assigned by us and cannot change
			return nil
		}
```

or you mean there is no point in checking node.Annotations[util.OvnTransitSwitchPortAddr] because it would have failed and been on nodeSyncFailed? I guess either way it doesn't hurt to have the check here.

Reply:
yep, the latter
All conditions of this function are already covered by other checks (including HO check in the very beginning of `handleAddUpdateNodeEvent`), and I find it confusing that it is only called on update and not add events. 
It can be also hard to maintain, e.g. if we add some annotation in the future, I am not sure if we will remember to update this function as well (and there is no need as long as we keep `nodeSyncFailed`)